src/VisualStudio/Core/Def/Storage/VisualStudioCloudCacheStorageService.cs

```diff
@@ -14,16 +18,35 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Storage
 {
+    /// <summary>
+    /// The storage service is a process wide singleton.  It will ensure that any workspace that
+    /// wants to read/write data for a particular solution gets the same DB.  This is important,
+    /// we do not partition access to the information about a solution to particular workspaces.
+    /// </summary>
+    [Export(typeof(VisualStudioCloudCacheStorageService)), Shared]
     internal class VisualStudioCloudCacheStorageService : AbstractCloudCachePersistentStorageService
     {
-        private readonly IAsyncServiceProvider _serviceProvider;
-
-        public VisualStudioCloudCacheStorageService(IAsyncServiceProvider serviceProvider, IPersistentStorageConfiguration configuration)
-            : base(configuration)
+        [ExportWorkspaceService(typeof(ICloudCacheStorageServiceFactory), ServiceLayer.Host), Shared]
+        internal class VisualStudioCloudCacheStorageServiceFactory : ICloudCacheStorageServiceFactory
         {
-            _serviceProvider = serviceProvider;
+            private readonly VisualStudioCloudCacheStorageService _service;
+
+            [ImportingConstructor]
+            [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+            public VisualStudioCloudCacheStorageServiceFactory(VisualStudioCloudCacheStorageService service)
+                => _service = service;
+
+            public IChecksummedPersistentStorageService Create()
+                => _service;
         }
```

Comment:
the workspace service allows features to then grab the singleton *through* a workspace service (which just returns the singleton).  This allows features to still just use Workspace/Solution/HostServices as their currency, without having to pull in the singleton through some other mechanism that they'd then have to pass along.  LMK if you're ok with this @tmat

Reply:
Conceptually it's fine, it's what we are doing elsewhere as well. I'd just rename this to -Provider and Create -> Get. Since it is not creating anything.

Reply:
true.  will do.